backend/capellacollab/core/authentication/basic_auth.py

```diff
@@ -0,0 +1,56 @@
+# SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+# SPDX-License-Identifier: Apache-2.0
+
+import datetime
+import logging
+
+import fastapi
+from fastapi import security, status
+
+from capellacollab.core import database
+from capellacollab.users import crud as user_crud
+from capellacollab.users.tokens import crud as token_crud
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPBasicAuth(security.HTTPBasic):
+    async def __call__(  # type: ignore
+        self, request: fastapi.Request
+    ) -> security.HTTPBasicCredentials | None:
+        credentials: security.HTTPBasicCredentials | None = (
+            await super().__call__(request)
+        )
+        if not credentials:
+            if self.auto_error:
+                raise fastapi.HTTPException(
+                    status_code=status.HTTP_401_UNAUTHORIZED,
+                    detail="Not authenticated",
+                    headers={"WWW-Authenticate": "Basic"},
+                )
+            return None
+        with database.SessionLocal() as session:
+            user = user_crud.get_user_by_name(session, credentials.username)
```

Comment:
we do through the "Token invalid for user" Exception if that is what you meant if the user is None l. 38

Reply:
I think when I reviewed this the `if user` check to get the `token_data` was not there yet. Therefore, the `user.id` to get the `token_data` produced a valid mypy error on my side. But with l. 35 we indeed through an error in case the user is None

Reply:
alright good 👍